### PR TITLE
add mas secrets

### PR DESCRIPTION
**Pull request by southeo**
add mas secrets

### CODE REVIEW

File: test/secret-manager/mas-secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: mas-secrets
+  namespace: default
```
Comment (samleeflang) [inline test/secret-manager/mas-secrets.yaml:5]: mas run in their own namespace

**Review by samleeflang — COMMENTED** [review]
MAS namespace needs SA which can access the secrets. Translator namespace has this already look there for inspiration.

**Comment by southeo** [comment]
we have the translator secrets separate from the default secrets, so i moved the mas secrets to the mas directory

File: test/machine-annotation-service/mas-service-account.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  name: mas-secret-manager
+  namespace: machine-annotation-services
+  annotations: {
+    eks.amazonaws.com/role-arn: "arn:aws:secretsmanager:eu-west-2:824841205322:secret:mas-secrets-QU1dOe"
+  }
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: read-secrets-global
```
Comment (samleeflang) [inline test/machine-annotation-service/mas-service-account.yaml:13]: This name is specific to the mas-secrets.
We could either create a generic ClusterRolBinding and update each time we add subjects.
Or create a different ClusterRoleBinding for each secret manager (as we have now) but then the name should indicate for which secret manager the binding is for

**Review by samleeflang — COMMENTED** [review]
Almost there

File: test/machine-annotation-service/mas-service-account.yaml (hunk 2 above)
Comment (southeo) [reply test/machine-annotation-service/mas-service-account.yaml]: So you're saying the name for the binding needs to be specific to the mas. Does the name need to be used anywhere else? 

The translator SA and the general SA have the same binding name. Does this need to change?

Comment (samleeflang) [reply test/machine-annotation-service/mas-service-account.yaml]: Probably needs to change. I don't think the name needs to be used anywhere else. The binding just binds the SA and the ClusterRole together.

**Review by samleeflang — APPROVED** [review]

**Pull request merged**